src/comics/comic_comments/routes.rs

```diff
@@ -166,48 +159,69 @@ pub async fn create_comment(
     State(pool): State<Pool<AsyncPgConnection>>,
     Path(comic_id): Path<Uuid>,
     Json(payload): Json<CreateComicComment>,
-) -> Result<(), ComicCommentsError> {
+) -> Result<Json<ComicCommentResponse>, ComicCommentsError> {
     let mut db = pool.get().await?;
 
-    db.transaction::<_, ComicCommentsError, _>(|transaction| {
-        async move {
-            let comment = ComicComment {
-                id: Uuid::now_v7(),
-                content: payload.content,
-                created_at: Utc::now(),
-                updated_at: None,
-                comic_id,
-                user_id: auth.current_user.id,
-            };
-
-            let comment = diesel::insert_into(comic_comments::table)
-                .values(&comment)
-                .get_result::<ComicComment>(transaction)
-                .await?;
-
-            if let Some(parent_comment_id) = payload.parent_comment_id {
-                let count = comic_comments_mapping::table
-                    .filter(comic_comments_mapping::parent_comment_id.eq(parent_comment_id))
-                    .select(diesel::dsl::count(comic_comments_mapping::child_comment_id))
-                    .get_result::<i64>(transaction)
+    let comment = db
+        .transaction::<_, ComicCommentsError, _>(|transaction| {
+            async move {
+                let comment = ComicComment {
+                    id: Uuid::now_v7(),
+                    content: payload.content,
+                    created_at: Utc::now(),
+                    updated_at: None,
+                    comic_id,
+                    user_id: auth.current_user.id,
+                };
+
+                let comment = diesel::insert_into(comic_comments::table)
+                    .values(&comment)
+                    .get_result::<ComicComment>(transaction)
                     .await?;
 
-                tracing::debug!("count: {count}");
-
-                diesel::insert_into(comic_comments_mapping::table)
-                    .values((
-                        comic_comments_mapping::parent_comment_id.eq(parent_comment_id),
-                        comic_comments_mapping::child_comment_id.eq(comment.id),
-                    ))
-                    .execute(transaction)
-                    .await?;
+                if let Some(parent_comment_id) = payload.parent_comment_id {
+                    let count = comic_comments_mapping::table
+                        .filter(comic_comments_mapping::parent_comment_id.eq(parent_comment_id))
+                        .select(diesel::dsl::count(comic_comments_mapping::child_comment_id))
+                        .get_result::<i64>(transaction)
+                        .await?;
+
+                    tracing::debug!("count: {count}");
```

Comment:
is this only for debugging?

Reply:
I added it a while ago to try and limit the nesting, but it can be removed it's useless

Reply:
yeah we should remove it for now